Issue 380 - added visualization showing products of ingested sources files

dickmancj-mil opened the pull request:
Visualization shows ingested source file on left side of graph, jobs that used it as input and outputs produced from those jobs.

wardcr commented:
Visualization looked reasonable to me with the static data.

ericsvendsen-mil commented:
A couple notes after looking at this with "real" data:

* Some ingest files do not have any associated detail (anything that's queued, but also sometimes even files that have been ingested).  When this happens, the api returns `{"detail":"Not found."}`. This should be accounted for in the UI.
* For files with a lot of jobs and products, the tree visualization is too tall to be really useful, since you need to scroll up and down so often just to see everything.
* It pushes the table below way far down on the page.
* When you do try to scroll to see everything, the page doesn't scroll when the mouse cursor is hovering over the tree view for some reason.
* My initial thought of getting around these issues would be to make the tree collapsible (and be collapsed by default) and possibly putting it in its own modal window.  If it's inline with other DOM elements, the constant resizing due to collapsing/expanding nodes could get tiresome.

Overall it's close, but we need to mock up a more robust static example in order to resolve these problems.

dickmancj-mil commented:
I think I've addressed all of the issues @ericsvendsen-mil identified. Added more robust test data, resizing graph container to fit the specified viewbox, and allowed scrolling event to pass through to the page.
